packages/core/src/common/crypto.ts

```diff
@@ -0,0 +1,14 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { isWeb } from './webUtils'
+
+export function randomUUID(): `${string}-${string}-${string}-${string}-${string}` {
+    if (isWeb()) {
+        return globalThis.crypto.randomUUID()
+    }
```

Comment:
Can we just use `globalThis.crypto` everywhere, and only "polyfill" the nodejs case. That avoids the need for this `isWeb()` check. We only need something like:

```
if (!isWeb()) {
    globalThis.crypto = ...
}